pkg/reconciler/broker/resources/pods.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// volumeGenerationKey is the annotation key for broker data plane pods whose
+// value is updated upon broker-targets configmap update. This is used to
+// trigger immediate configmap volume refresh without restarting the pod.
+const volumeGenerationKey = "volumeGeneration"
+
+// UpdateVolumeGeneration updates the volume generation annotation on the pod to
+// trigger volume mount refresh (for configmap).
+func UpdateVolumeGeneration(kubeClientSet kubernetes.Interface, p *corev1.Pod) error {
+	pod := p.DeepCopy()
+	annotations := pod.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string, 1)
+	}
+	gen, _ := strconv.Atoi(annotations[volumeGenerationKey])
+	annotations[volumeGenerationKey] = strconv.Itoa(gen + 1)
+	pod.SetAnnotations(annotations)
```

Comment:
Is it possible (or does it matter) if this generation differs from each other? An alternative is to have the controller to write the config generation as a configmap annotation and use it for pod annotation (instead of reading each pod for the previous value). This might also appear useful when check if a pod has the latest config loaded or not.

Reply:
We only need an update on the annotation so actual value doesn't matter. But I like the other point that we can compare pod annotation with configmap annotation.

Added an issue https://github.com/google/knative-gcp/issues/913

Reply:
I almost tried to do it in this PR. But not quite sure how to test the configmap generation logic with the way the controller tests are setup. Need some more investigation. I will do it as a follow up ASAP.

Reply:
Sg.

Reply:
I don't know how we should handle the annotation for new pods though.

Reply:
I don't think it's very important for new pods to have the annotation. New pods should get the latest configmap anyway.

For debugging purpose, we can put the generation number as part of the targets config. So we can still find the generation number from within the configmap volume of a new pod.